packages/pathfinder/src/pathfinder.ts

```diff
@@ -0,0 +1,400 @@
+const config =  require('../../../config.json');
+const { Pool } =  require('pg');
+const pool = new Pool({
+    user: 'postgres',
+    host: 'interstellar.market',
+    database: 'silent_shop',
+    password: config.pg.password,
+    port: 5432,
+});
+
+
+export interface stellarAsset {
+    code: string;
+    issuerPublicKey: string;
+}
+
+export interface IPathInfo {
+    to: stellarAsset;
+    from: stellarAsset;
+    path: stellarAsset[] | null;
+    bucket: number;
+    exchangeRate: number | null; // to / from
+}
+
+interface stellarBalance { 
+    balances: Array<
+    {
+        balance: string
+        asset_type: 'native'
+    } |
+    {
+        balance: string
+        limit: string
+        asset_type: 'credit_alphanum4' | 'credit_alphanum12'
+        asset_code: string
+        asset_issuer: string
+    }
+    >;
+}
+
+interface Cache {
+    cache: any;
+    savePathsToCache(key: string, bucket: number, bestPaths: IPathInfo[]): any;
+    removeFromCache(key: string): any;
+    lookupInCache(key: string): any;
+}
+
+export class PathfinderInitializationError extends Error {}
+
+import * as StellarSdk from 'stellar-sdk'; 
+
+export class Pathfinder {
+
+    /**
+     * This public key should correspond to an account that holds balances of each of
+     * the assets that we support. It will be used as both the destination and source
+     * account for all of the pathfinding algorithms. TODO fill it in
+     */
+    public static REFERENCE_ACCOUNT_PUBLIC_KEY = "GBG47DCYZCY4UU4UN7XSECYT45IRTCUGZN73SVZE5TSCIB3DQDREYUTS";
+    public supportedAssets: stellarAsset[];
+    private server: StellarSdk.Server;
+
+    constructor() {
+        this.server = new StellarSdk.Server('https://horizon.stellar.org');
+    }
+
+    /****************************************
+     * Initialization methods
+     ****************************************/
+
+    /**
+     * Should perform the following actions:
+     *
+     * - loads the list of supported assets
+     */
+    public async init() {
+
+        return Promise.all([
+            this.getSupportedAssets().then(() => this.validateSupportedAssets(this.supportedAssets))
+        ]);
+
+    }
+
+    /**
+     * Loads the list of supported assets from the postgres database
+     */
+    private async getSupportedAssets() {
+
+        let assets = [] as stellarAsset[];
+        const query = {
+            text: 'SELECT id, asset_code, asset_issuer, asset_type from ASSETS'
+        };
+        return pool.query(query)
+            .then((queryResult: any) => queryResult.rows)
+            .then((dbAssets: any) => {
+                assets = dbAssets;
+                this.supportedAssets = dbAssets;
+                return dbAssets;
+            });
+            
+    }
+
+    /**
+     * helper for below
+     * returns a key to be used for lookup in the form of `code-issuer`
+     */
+    private makeKeyPairHelper = (asset_code: string, asset_issuer: string, asset_type: string): string => {
```

Comment:
All of the stuff that interacts with the stellar sdk (e.g., creating assets) should be in the stellar section.

Reply:
gotcha